### PR TITLE
More board logic

### PR DESCRIPTION
**Pull request by sulphite**
adds:

- new tests 
- `getRow` function to create rows
- `isPushable` to check if a row can be pushed into
- documentation comments in `Board`

### CODE REVIEW

File: src/tests/board.test.ts [hunk 1]
```diff
@@ -32,3 +32,25 @@ test("given two neighboring hexes we can get the direction from A to B", () => {
   const direction = board.findDirection([0, 0, 0], [0, 1, -1]);
   expect(direction).toBe(Direction.SE);
 });
+
+test("given an outer tile and a neighboring inner tile I can get a row of only the inner tiles", () => {
+  const row = board.getRow([4, 0], [3, 0]);
+  console.log("row length", row.size);
```
Comment (adrianHards) [inline src/tests/board.test.ts]: better to remove console.logs

File: src/tests/board.test.ts [hunk 2]
```diff
@@ -32,3 +32,25 @@ test("given two neighboring hexes we can get the direction from A to B", () => {
...
+  expect(row.filter((tile) => !tile.isOuterTile()).size).toEqual(row?.size);
+});
+
+test("given an outer tile not at the corner I can get a row of only the inner tiles", () => {
+  const row = board.getRow([4, -1], [3, 0]);
+  console.log("row length", row.size);
+  expect(row.filter((tile) => !tile.isOuterTile()).size).toEqual(row?.size);
+});
+
+test("given a row isPushable returns true if there is at least one empty space", () => {
```
Comment (adrianHards) [inline src/tests/board.test.ts]: add a test where tile isn't push-able?

File: src/tests/board.test.ts [hunk 3]
```diff
@@ -32,3 +32,25 @@ test("given two neighboring hexes we can get the direction from A to B", () => {
...
+test("given an outer tile and a neighboring inner tile I can get a row of only the inner tiles", () => {
+  const row = board.getRow([4, 0], [3, 0]);
+  console.log("row length", row.size);
+  expect(row.filter((tile) => !tile.isOuterTile()).size).toEqual(row?.size);
+});
+
+test("given an outer tile not at the corner I can get a row of only the inner tiles", () => {
+  const row = board.getRow([4, -1], [3, 0]);
+  console.log("row length", row.size);
+  expect(row.filter((tile) => !tile.isOuterTile()).size).toEqual(row?.size);
```
Comment (adrianHards) [inline src/tests/board.test.ts]: perhaps more readable to define expected and observed first, e.g. on a line expectedRowLength = row.size and then use within toEqual, as an example

**Review by adrianHards — CHANGES_REQUESTED** [review]
Are all possible edge cases accounted for? I'll need help understanding how it works. Recommend writing out possible test cases first.

File: src/logic/Board.ts [hunk 4]
```diff
@@ -20,14 +21,23 @@ export class Board implements IBoard {
     this.grid = new Grid(Tile, spiral({ radius: 4 }));
   }
 
+  /**
+   * returns all neighbour tiles of a coordinate, within grid limits
+   * @param coord a hexcoordinate
```
Comment (adrianHards) [inline src/logic/Board.ts]: I like the idea, but not sure this is particularly helpful, example values might help. Might also help to agree on standardized terms and use throughout.

**Review by adrianHards — CHANGES_REQUESTED** [review]

File: src/tests/board.test.ts (hunk 3 above)
Comment (sulphite) [reply src/tests/board.test.ts]: it is done

**Comment by adrianHards** [comment]
@sulphite should vector type and predefined vectors be kept in board class? Latter might be fine, unless used elsewhere.

File: src/logic/Board.ts [hunk 5]
```diff
@@ -20,23 +26,60 @@ export class Board implements IBoard {
...
-  findDirection(coordA: HexCoordinates, coordB: HexCoordinates) {
+  /**
+   * Helper method to find direction based on vectors.
+   * Searches a predefined array of vectors to match input vectors to a direction.
+   * @param q - q vector component
+   * @param r - r vector component
+   * @param s - s vector component
+   * @returns The matching direction if found, throws an error otherwise.
+   */
+  getDirectionByVectors(q: number, r: number, s: number): Direction {
```
Comment (adrianHards) [inline src/logic/Board.ts]: feel free to revert to findDirection

**Review by adrianHards — APPROVED** [review]
